pkg/cloud/services/ec2/subnets.go

```diff
@@ -342,6 +399,8 @@ func (s *Service) getSubnetTagParams(id string, public bool, manualTags infrav1.
 	name.WriteString(s.scope.Name())
 	name.WriteString("-subnet-")
 	name.WriteString(role)
+	name.WriteString("-")
+	name.WriteString(strings.Replace(zone, "-", "", -1))
```

Comment:
I'm curious about this one – do we bump up against length limits, otherwise?

Reply:
I haven't hit any limits, it was really just to reduce the the number of -.
Looking at the docs the limit on the value is 256 characters
https://docs.aws.amazon.com/general/latest/gr/aws_tagging.html

Reply:
Oh, in that case I have a mild preference for leaving the `-`s in, to be consistent with what we do elsewhere in the failureDomains and network.availabilityZones portions of the AWSCluster status.

Reply:
Changed so the `-` are left in on the subnet and route table names.